Source/WebCore/style/StyleBuilderConverter.h

```diff
@@ -2174,7 +2174,14 @@ inline OptionSet<MarginTrimType> BuilderConverter::convertMarginTrim(BuilderStat
     if (!list || !list->size())
         return RenderStyle::initialMarginTrim();
     OptionSet<MarginTrimType> marginTrim;
-    ASSERT(list->size() <= 4);
+    for (auto& item : *list) {
+        if (item.valueID() == CSSValueBlock)
+            marginTrim.add({ MarginTrimType::BlockStart, MarginTrimType::BlockEnd });
+        if (item.valueID() == CSSValueInline)
+            marginTrim.add({ MarginTrimType::InlineStart, MarginTrimType::InlineEnd });
+    }
+    if (!list->size())
+        return marginTrim;
```

Comment:
Isn't this already handled a few lines above?

Reply:
Ah I meant to write `if (!marginTrim->isEmpty())` :)